src/app/dashboard/store-form.tsx

```diff
@@ -13,32 +13,54 @@ import {
   FormMessage,
 } from "@/components/ui/form";
 import { Button } from "@/components/ui/button";
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
+import { AlertCircle } from "lucide-react";
 import { MerchantInfo } from "../dto/merchant";
+import { UploadButton } from "@/utils/uploadthing";
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import Image from "next/image";
+import { Card, CardFooter, CardHeader } from "@/components/ui/card";
 
 const formSchema = z.object({
-  storeName: z.string().min(2).max(50),
-  storeUrl: z.string().url().optional(),
-  storeDescription: z.string().min(2).max(100),
-  storeLogo: z.string().min(0),
+  name: z.string().min(2).max(50),
+  url: z.string().url().optional(),
+  description: z.string().min(2).max(100),
+  profileImage: z.string().min(0).optional(),
 });
 
 export function StoreForm() {
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [fileKey, setFileKey] = useState<string | undefined>(undefined);
+  const { push } = useRouter();
+
   const form = useForm<z.infer<typeof formSchema>>({
     resolver: zodResolver(formSchema),
   });
 
+  const removePreview = (res: any) => {
+    fetch(`${process.env.NEXT_PUBLIC_SERVER_URL}api/merchant/removeImage/`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        imagePath: fileKey,
+      }),
+    });
+    form.setValue("profileImage", undefined);
+  };
+
   // 2. Define a submit handler.
   async function onSubmit(values: z.infer<typeof formSchema>) {
     const merchantInfo: MerchantInfo = {
-      storeName: values.storeName,
-      storeDescription: values.storeDescription,
-      storeUrl: values.storeUrl,
+      name: values.name,
+      description: values.description,
+      url: values.url,
+      profileImage: values.profileImage,
     };
-    console.log(
-      `${process.env.NEXT_PUBLIC_SERVER_URL}${process.env.NEXT_PUBLIC_SAVE_MERCHANT_PATH}`
-    );
     const response = await fetch(
-      `${process.env.SERVER_URL}${process.env.SAVE_MERCHANT_PATH}`,
+      `${process.env.NEXT_PUBLIC_SERVER_URL}${process.env.NEXT_PUBLIC_SAVE_MERCHANT_PATH}`,
```

Comment:
il save merchant path cambia in base all'env? Senno' possiamo metterlo tipo in un file di constanti

Reply:
No. Usiamo il file di costanti meglio. hai ragione